### PR TITLE
feat: optionally use token from env, allow selecting repos

### PR DESCRIPTION
**Pull request by sammcj**
- When prompting the user for a token give the option of using a token from the env ($GITHUB_TOKEN)
- Allow selecting all, or some repositories

### CODE REVIEW

**Comment by sammcj** [comment]
Then if this gets merged - this might be another little fix - https://github.com/domdomegg/github-repo-settings-editor/pull/3

File: src/index.ts
Comment (domdomegg) [inline src/index.ts]:
```suggestion
    validate: (value: string) =>
```

This should remain `string`
https://stackoverflow.com/questions/14727044/what-is-the-difference-between-types-string-and-string

File: src/index.ts
Comment (domdomegg) [inline src/index.ts]: Maybe we should show this option (and entire question?) if there is a string beginning `ghp_` in `process.env.GITHUB_TOKEN`?

Cost of this as a proposed approach is that it reduces the discoverability of this as an option, but I think that's an okay trade-off. We could always add a log line somewhere else along the lines of `Tip: Want to avoid entering your token next time? You can set it in the environment variable GITHUB_TOKEN.`

File: src/index.ts
Comment (domdomegg) [inline src/index.ts:86]: nit: Usually you'd put the imports at the top of the file, rather than dynamically importing at this point. But I can see the argument for not trying to import them if you're not reading files.

File: src/index.ts
Comment (domdomegg) [inline src/index.ts:90]: We should check the file is well formatted (both in terms of being valid JSON, and actually being an array of `Repository` objects), and throwing a useful error if not.

File: src/index.ts
Comment (domdomegg) [inline src/index.ts:143]: nit: formatting, probably want a newline somewhere in here

File: src/index.ts
Comment (domdomegg) [inline src/index.ts:142]: nit: It's a bit odd to have this function nested within the other. It might be simpler to extract the filtering part to an extra step that is just called directly from the main function near the bottom (where `getRepositories` itself is called)

File: src/index.ts
Comment (domdomegg) [inline src/index.ts:84]: `grse-config.json` might be a better name?

The structure should probably not just be `Repository[]`, as otherwise you can't specific your desired preferences in future. Maybe it being an object containing a `repos: Repository[]` entry, so it is extensible with the options you want to set etc. (i.e. the `Preferences` type).

**Review by domdomegg — CHANGES_REQUESTED** [review]
Thanks for the PR, and sorry for the barrage of PR comments. Hopefully it's helpful feedback.

General framework I use to approve things is that if:
- It doesn't expose broken features
- It doesn't commit us to going down a bad path
- It doesn't significantly reduce code readability or structure
I'll approve it.

Nits are explicitly things that don't fall into the above. Other things also might not fit in to the comments above (usually quantified with 'Maybe', 'Probably', 'In future'), so I'll still approve but they should probably be at least considered and may need fixing at some future point.

Once a PR is approved, I'm happy for you to merge it or make additonal changes. Please shout if you don't have permissions to actually do the merge!

**Comment by sammcj** [comment]
Totally agree with all your comments @domdomegg , thanks so much for taking the time to write them, I'll cut up a cleaned up version soon 😄

File: src/index.ts
Comment (sammcj) [reply src/index.ts]: Good advice, thank you - implemented this one.

**Comment by sammcj** [comment]
closing for now as have become insanely busy all of a sudden, might completely redo this in the future.

**Pull request closed without merging**
